### PR TITLE
Update SeriesAPI to take account of prison id and prison categories

### PR DESCRIPTION
**Pull request by psoleckimoj**

### CODE REVIEW

File: modules/custom/moj_resources/src/SeriesContentApiClass.php [hunk 1]
```diff
@@ -195,44 +211,179 @@ private function translateNode(NodeInterface $node)
...
+      ->condition('field_prison_categories', $prison_categories, 'IN');
+    $no_prison_categories_filter = $query
+      ->andConditionGroup()
+      ->notExists('field_moj_prisons')
+      ->notExists('field_prison_categories');
+    $content_filter = $query
+      ->orConditionGroup()
+      ->condition('field_moj_prisons', $prison_id, '=')
+      ->condition($prison_categories_filter)
+      ->condition($no_prison_categories_filter);
```
Comment (spikeheap) [inline modules/custom/moj_resources/src/SeriesContentApiClass.php:319]: Is this needed now we've back-filled categories to existing series?

File: modules/custom/moj_resources/src/SeriesContentApiClass.php [hunk 2]
```diff
@@ -19,31 +20,42 @@ class SeriesContentApiClass
    *
    * @var array
    */
-  protected $nids = array();
+  protected $node_ids = array();
```
Comment (spikeheap) [inline modules/custom/moj_resources/src/SeriesContentApiClass.php:23]: 🤩 loving the switch to more meaningful array names ❤️

Comment (LucasCairns) [reply modules/custom/moj_resources/src/SeriesContentApiClass.php:23]: Definitely helps, there was a variable named `results` which turned out to be a `query`, something we should be mindful of as we go forward is fixing these variable names

Comment (psoleckimoj) [reply modules/custom/moj_resources/src/SeriesContentApiClass.php:23]: It was a good call by @LucasCairns to do that 😄 We need to do it across all the code.

**Review by LucasCairns — COMMENTED** [review]
Closing to change merge branch

**Pull request closed without merging**